service/history/decisionHandler.go

```diff
@@ -576,6 +548,25 @@ Update_History_Loop:
 			}
 
 			return nil, updateErr
+		} else {
+			// at this point the update is successful, so answer all the queries which we have answers for
```

Comment:
Also move the section to a separate method. This method is already way too long.

Reply:
https://github.com/uber/cadence/issues/2740

I do not want to refactor this method piece by piece. I think we need to take a full pass over this method so we can refactor in a consistent manner. For now I prefer to leave it as one chunk of code.

Reply:
I agree this method needs to be refactored but I think doing it piece wise and then refactoring it will be harder.